todo_app/app.py

```diff
@@ -210,7 +233,8 @@ def login_callback():
 
         request_text = requests.get(url, headers=headers, data=body).text
 
-        json_data = json.loads(request_text)
+        json_data = response.json(request_text)
```

Comment:
This change breaks your authentication for me; when making a change like this it's worth checking it works (e.g. by opening your app in private-browsing, or clearing your cookies). The reason is that the `response` variable Ben was referring to is assumed to be the result of your request rather than the response object that has been imported from urllib:
```
from urllib import parse, response
```

Instead, if using this pattern it would look more like:
```python
        response = requests.get(url, headers=headers, data=body)
        json_data =  response.json()
```
In which case you could remove the import from urllib for `response` too